base/range.jl

```diff
@@ -6,9 +6,9 @@ colon{T<:Real}(start::T, stop::T) = UnitRange{T}(start, stop)
 
 range(a::Real, len::Integer) = UnitRange{typeof(a)}(a, oftype(a, a+len-1))
 
-colon{T}(start::T, stop::T) = colon(start, oftype(stop-start, 1), stop)
+colon{T}(start::T, stop::T) = colon(start, oneunit(stop-start), stop)
```

Comment:
Sorry I didn't look at this earlier. I think the original version was correct, and that this version is a bug. This has been discussed elsewhere and there isn't a uniform opinion, but I would find the following worrisome:
```julia
xmm, ymm = 1000mm, 2000mm
xm, ym = convert(Meter, xmm), convert(Meter, ymm)
xmm == xm && ymm == ym => true
(xmm:ymm) == (xm:ym) => false
```
I think it's a bad idea to allow `start:stop` constructors for unitful quantities; one should be required to specify the `step` explicitly. The `oftype` version of this code forces that.